### PR TITLE
Feat/ver.3.0.0

### LINKED ISSUE #23: migrate to latest version & publish this package as v3
- update every dependencies of this project
- update rules if above update requires it
- run tests (see package.json) to make sure whether it works well

### PR DESCRIPTION
**Pull request by 4dvn-hieu**
- Updated all dependencies to latest versions.
- `babel-eslint` is deprecated & replaced by `@babel/eslint-parser`. `@babel/eslint-parser` requires other `@babel` libraries and a config file (`babel.config.js`) to work.
- eslint 8 caused a breaking change to plugins (breaks `eslint-plugin-dollar-sign`) so I had to fork `eslint-plugin-dollar-sign`, fixed it and published it to npm. Then I changed the dependency to `@4dvn-hieu/eslint-plugin-dollar-sign`.
- eslint 8 replaced CLIEngine object with ESLint object so I modified spec.js files to make them work.

### CODE REVIEW

**Comment by TakuyaAbe** [comment]
@4dvn-hieu thanks! I'll take a look in this week

File: package.json [hunk 1]
```diff
@@ -27,17 +27,20 @@
...
+    "eslint-config-prettier": "^8.5.0",
+    "@4dvn-hieu/eslint-plugin-dollar-sign": "^1.0.3",
+    "eslint-plugin-prettier": "^4.0.0",
+    "eslint-plugin-react": "^7.29.4",
+    "prettier": "^2.6.1"
   },
   "devDependencies": {
-    "jest": "^26.4.2",
-    "react": "^16.13.1"
+    "jest": "^27.5.1",
```
Comment (4dvn-hoangvu) [inline package.json:43]: I noticed this PR is quite long ago. Jest has released another major upgrade. 
Could you please upgrade the jest package to the latest version 🙇

**Review by 4dvn-hoangvu — CHANGES_REQUESTED** [review]
@4dvn-hieu 
There is only one package which is not up-to-date. Could you check it in my comment?

**Review by 4dvn-hoangvu — APPROVED** [review]
@4dvn-hieu LGTM 📦

**Pull request merged**